build_tools/waf_dynamo.py

```diff
@@ -490,6 +492,11 @@ def default_flags(self):
             'MIN_SAFARI_VERSION=90000',
             'MIN_CHROME_VERSION=32']
 
+        if Options.options.with_webgpu and platform_supports_feature(build_util.get_target_platform(), 'webgpu', {}):
+            emflags_link += ['USE_WEBGPU', 'GL_WORKAROUND_SAFARI_GETCONTEXT_BUG=0']
+            # This is needed so long as we have to use sleep to make initialization blocking
+            emflags_link += ['ASYNCIFY', 'ASYNCIFY_ADVISE', 'ASYNCIFY_IGNORE_INDIRECT', 'ASYNCIFY_ADD=["main", "dmEngineCreate(int, char**)"]' ]
```

Comment:
I had just set it to asyncify everything and it was pretty huge, late last week I added this restriction on only doing the few functions we need (as the only async operation is on initialization flow). I think medium term this is acceptable, current webgpu.h has some future concept in it that I think will make it possible to use instead of this sledgehammer, if that doesn't pan out it might be worth considering an async graphics initialization flow. I actually started on doing that before I discovered I could use asyncify if you are interested in that patch I can get it for you.